packages/tanstack-react-query/src/internals/createOptionsProxy.ts

```diff
@@ -46,9 +66,19 @@ export interface DecorateQueryProcedure<TDef extends ResolverDef> {
    * @see https://tanstack.com/query/latest/docs/framework/react/reference/infiniteQueryOptions#infinitequeryoptions
    */
   infiniteQueryOptions: TRPCInfiniteQueryOptions<TDef>;
+
+  /**
+   * Useful for query invalidation
+   *
+   * @see https://tanstack.com/query/latest/docs/framework/react/guides/query-invalidation
+   */
+  getQueryKey: (input?: TDef['input']) => TRPCQueryKey;
```

Comment:
^ @juliusmarminge

Reply:
yea but the Omit just leaves `invalite` in there whose type is wrong. 

Is this a future-proofing or else you could just inline both since they'll both be separate from the base type you're extending

Reply:
Oh sorry you're right, yeah I need to do the same thing here as I did for getQueryKey

Probably just need inlining, but would like to figure out the right API before going too hard here